packages/unlock-js/src/PublicLock/v7/setKeyManagerOf.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Change lock manager for a specific key
+ * @param {string} lockAddress address of the lock
+ * @param {string} managerAddress address of the user
+ */
+
+export default async function (
+  { lockAddress, managerAddress, tokenId },
+  transactionOptions = {},
+  callback
+) {
+  const lockContract = await this.getLockContract(lockAddress)
+  const transactionPromise = lockContract.setKeyManagerOf(
+    tokenId,
+    managerAddress
+  )
+
+  const hash = await this._handleMethodCall(transactionPromise)
+
+  if (callback) {
+    callback(null, hash)
+  }
+
+  await this.provider.waitForTransaction(hash)
```

Comment:
I do wonder if we should move away from this to something like returning `{ hash, wait: this.provider.waitForTransaction(hash) }`

This way, the code could be

```ts
const tx = await walletService.setKeyManager()
console.log(tx.hash)
const confirmation = await tx.wait()
```

Reply:
I see also this approach in other parts, is useful when we need to debug or there is more.?

Reply:
For now let's favor consistency please @searchableguy ... we can change the overall approach in the library too but if we do it we need to do it everywhere!